ed25519/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+extern crate ed25519_dalek;
+extern crate ewasm_api;
+extern crate sha2;
+
+mod verify;
+
+#[cfg(not(test))]
+#[no_mangle]
+pub extern "C" fn main() {
+    let length = ewasm_api::calldata_size();
+
+    ewasm_api::consume_gas(2000);
+
+    if length < 128 {
+        ewasm_api::revert();
+    }
+
+    // FIXME: use ewasm-rust-api 0.6.x to have slices
+    let input = ewasm_api::unsafe_calldata_copy(0, length);
+    let mut tmp = [0u8; 128];
+    tmp.copy_from_slice(&input);
```

Comment:
If `length` > 128, then this will end up truncating the input data. Is this the intended behavior?

Reply:
Looks like parity's impl of the precompile truncates the input data as well.

Reply:
Yes.  Based on how it is implemented in geth: https://github.com/ethereum/go-ethereum/pull/16453/files

Reply:
There's a parity implementation?

Reply:
in native

Reply:
https://github.com/paritytech/parity-ethereum/pull/8330/files @axic

Reply:
never got merged, i guess. stale